source/opt/loop_unroller.cpp

```diff
@@ -99,13 +100,18 @@ struct LoopUnrollState {
         new_condition_block(nullptr),
         new_header_block(nullptr) {
     previous_phis_ = std::move(phis);
+    previous_phi_value_ids_ = std::move(phi_initial_value_ids);
   }
 
   // Swap the state so that the new nodes are now the previous nodes.
   void NextIterationState() {
     previous_phi_ = new_phi;
     previous_latch_block_ = new_latch_block;
     previous_condition_block_ = new_condition_block;
+    for (size_t index = 0; index < previous_phis_.size(); ++index) {
+      previous_phi_value_ids_[index] =
```

Comment:
I suggest using a `map` here.

Reply:
Do you mean using a `map` for `previous_phi_value_ids_`?
`previous_phi_value_ids_` specify the value of phi for the previous loop iteration. It must use the same index as `previous_phis_`. Since `previous_phis_` is a vector, we should use a vector for it. Using `map` will make the code complicated.